src/text_statistics/TextStatistics.java

```diff
@@ -0,0 +1,66 @@
+package text_statistics;
+
+public class TextStatistics {
+    public static int countCharsWithSpaces(String fromUser){
+        int count = 0;
+        for (int i = 0; i < fromUser.length(); i++) {
+            if (Character.isLetter(fromUser.charAt(i)))
+                count++;
+            if (!Character.isLetter(fromUser.charAt(i)))
+                count++;
+        }
+        return count;
+    }
+    public static int countCharsWithoutSpaces(String fromUser){
+        int count = 0;
+        for(int i = 0; i < fromUser.length(); i++) {
+            if(fromUser.charAt(i) != ' ')
+                count++;
+        }
+
+        return count;
+    }
+    public static int countWords(String fromUser) {
+        if (fromUser == null || fromUser.isEmpty()) {
+            return 0;
+        }
+        int wordCount = 0;
+        boolean isWord = false;
+        int endOfLine = fromUser.length() - 1;
+        char[] characters = fromUser.toCharArray();
+
+        for (int i = 0; i < characters.length; i++) {
+            if (Character.isLetter(characters[i]) && i != endOfLine) {
+                isWord = true;
+            } else if (!Character.isLetter(characters[i]) && isWord) {
+                wordCount++;
+                isWord = false;
+            } else if (Character.isLetter(characters[i]) && i == endOfLine) {
+                wordCount++;
+            }
+        }
+        return wordCount;
+    }
+    public static void printValuesOfTextStatistics (String fromUser){
+        System.out.println("ilość znaków ze spacjami wynosi: " + countCharsWithSpaces(fromUser));
+        System.out.println("ilość znaków bez spacji wynosi: " + countCharsWithoutSpaces(fromUser));
+        System.out.println("ilość słów wynosi: " + countWords(fromUser));
+    }
+    public static boolean palindromeTest (String fromUser){
+        int length = fromUser.length();
```

Comment:
Tutaj brakuje mi wywołania funkcji `toLowerCase()` ponieważ `Ala` to jest palindrom a u Ciebie zwróci to niestety `false` z racji tego że `A` != `a`. Można to było zrobić prościej:
```
 String lowerText = text.toLowerCase();
        StringBuilder textToCalculate = new StringBuilder(lowerText);
        StringBuilder reverse = textToCalculate.reverse();
```